File: easy/0455_assign_cookies/solution.cpp
#include <vector>

class Solution {
public:
  int findContentChildren(std::vector<int> &g, std::vector<int> &s) {
    sort(g.begin(), g.end());
    sort(s.begin(), s.end());

    const int gLen = g.size();
    const int sLen = s.size();

    int gi = 0;
    int si = 0;

    while (gi < gLen && si < sLen) {
      if (g[gi] <= s[si]) {
        ++gi;
        ++si;
      } else
        ++si;
    }

    return gi;
  }
};

File: easy/0455_assign_cookies/solution.py
from typing import List


class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort(reverse=True)
        s.sort(reverse=True)

        cookies_count = 0

        while g and s:
            if g[-1] <= s[-1]:
                g.pop()
                s.pop()
                cookies_count += 1
            else:
                s.pop()

        return cookies_count
